src/App.jsx

```diff
@@ -14,45 +17,102 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_FIELD_BY_LETTERS = 'letters';
+const SORT_FIELD_BY_LENGTH = 'length';
+const IS_REVESER_DEFAULT_VALUE = false;
+
+function getPreparedGoods(goods, { sortField, reverseField }) {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((a, b) => {
+      switch (sortField) {
+        case SORT_FIELD_BY_LETTERS:
+          return a.localeCompare(b);
+        case SORT_FIELD_BY_LENGTH:
+          return a.length - b.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverseField) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
+function handlerResetClick(setSortField, setReverseField) {
+  setSortField('');
+  setReverseField(IS_REVESER_DEFAULT_VALUE);
+}
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [reverseField, setReverseField] = useState(IS_REVESER_DEFAULT_VALUE);
+  const visibleGoods = getPreparedGoods(
+    goodsFromServer,
+    { sortField, reverseField },
+  );
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          onClick={() => setSortField(SORT_FIELD_BY_LETTERS)}
+          type="button"
+          className={cn(
+            'button',
+            'is-info',
+            { 'is-light': sortField !== SORT_FIELD_BY_LETTERS },
+          )}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          onClick={() => setSortField(SORT_FIELD_BY_LENGTH)}
+          type="button"
+          className={cn(
+            'button',
+            'is-success',
+            { 'is-light': sortField !== SORT_FIELD_BY_LENGTH },
+          )}
+        >
+          Sort by length
+        </button>
+
+        <button
+          onClick={() => setReverseField(!reverseField)}
+          type="button"
+          className={cn(
+            'button',
+            'is-warning',
+            { 'is-light': !reverseField },
+          )}
+        >
+          Reverse
+        </button>
+
+        {(sortField || reverseField) && (
```

Comment:
It's better to create a variable for `sortField || reverseField` so you don't need to combine different logical operators